server/api.js

```diff
@@ -2,100 +2,416 @@ import { Router } from "express";
+		} else {
+			const query =
+				"UPDATE users SET first_name=$2, last_name=$3, region=$4, role=$5, email=$6, password=$7 WHERE id=$1 RETURNING *";
+			pool
+				.query(query, [
+					id,
+					first_name,
+					last_name,
+					region,
+					role,
+					email,
+					password,
+				])
+				.then(() =>
+					res.json({ success: "User has been Updated!", user: result.rows[0] })
+				)
+				.catch((err) => console.error(err.message));
+		}
 	});
 });
 
-router.get("/tasks", (_, res) => {
-	pool.query("SELECT * FROM tasks", (error, result) => {
-		res.json(result.rows);
-	});
+//DELETE a user with id
+router.delete("/users/:id", (req, res) => {
+	const { id } = req.params;
+	const query = "DELETE FROM users WHERE id=$1 RETURNING *";
+	pool
+		.query(query, [id])
+		.then((result) => {
+			if (result.rowCount) {
+				res.json({ success: "User has been deleted!", user: result.rows[0] });
+			} else {
+				res.status(404).json({ status: 404, msg: "Not found user!" });
+			}
+		})
+		.catch((err) => console.error(err.message));
+});
+
+// GET all plans form specific user
+router.get("/users/:user_id/plans", (req, res) => {
+	const { user_id } = req.params;
+	const query = "SELECT * FROM plans WHERE user_id=$1";
+	pool
+		.query(query, [user_id])
+		.then((result) => {
+			if (result.rowCount) {
+				res.json(result.rows);
+			} else {
+				res.status(404).json({ status: 404, msg: "Not found plans!" });
+			}
+		})
+		.catch((err) => console.error(err.message));
 });
 
-router.post("/register", async (req, res) => {
+//POST a plan for specific user id
+router.post("/users/:user_id/plans", (req, res) => {
+	const { user_id } = req.params;
+	const { description } = req.body;
+	const query =
+		"INSERT INTO plans (user_id, description ) VALUES ($1, $2) RETURNING *";
+	pool
+		.query(query, [user_id, description])
+		.then((result) => {
+			res.json({ success: "New plan has been created!", user: result.rows[0] });
+		})
+		.catch((err) => console.error(err.message));
+});
+
+// GET a plan form specific user with plan id
+router.get("/users/:user_id/plans/:plan_id", (req, res) => {
+	const { user_id, plan_id } = req.params;
+	const query = "SELECT * FROM plans WHERE id=$1 and user_id=$2";
+
+	pool
+		.query(query, [plan_id, user_id])
+		.then((result) => {
+			console.log(result);
+			if (result.rowCount) {
+				res.json(result.rows);
+			} else {
+				res.status(404).json({
+					status: 404,
+					msg: "This user doesn't have a plan with this id",
+				});
+			}
+		})
+		.catch((err) => console.error(err.message));
+});
+
+//UPDATE a plan details with user id
+router.put("/users/:user_id/plans/:plan_id", (req, res) => {
+	const { user_id, plan_id } = req.params;
+	const { description } = req.body;
+	pool
+		.query("SELECT * FROM plans WHERE user_id=$1 and id=$2", [user_id, plan_id])
+		.then((result) => {
+			if (result.rows.length === 0) {
+				return res.status(400).json("We couldn't find user id!");
```

Comment:
* 404
* _We couldn't find user id!_ is this really the issue? or does it have to do with both user and plan (I think the error message from the `GET` would work well here)